### PR TITLE
Update E0116 to new error code format.

### LINKED ISSUE #35249: E0116 needs to be updated to new format
From: src/test/compile-fail/E0116.rs

Error E0116 needs a span_label, an updated error text and a note, updating from:

```
error[E0116]: cannot define inherent `impl` for a type outside of the crate where the type is defined; define and implement a trait or new type instead
  --> src/test/compile-fail/E0116.rs:11:1
   |
11 | impl Vec<u8> {} //~ ERROR E0116
   | ^^^^^^^^^^^^^^^
```

To:

```
error[E0116]: cannot define inherent `impl` for a type outside of the crate where the type is defined 
  --> src/test/compile-fail/E0116.rs:11:1
   |
11 | impl Vec<u8> {} //~ ERROR E0116
   | ^^^^^^^^^^^^^^^ impl for type defined outside of crate
   |
   = note: define and implement a trait or new type instead
```

### PR DESCRIPTION
**Pull request by terryxsun**
Fixes #35249 as part of #35233.

r? @jonathandturner

### CODE REVIEW

File: src/librustc_typeck/coherence/orphan.rs [hunk 1]
```diff
@@ -33,10 +33,13 @@ struct OrphanChecker<'cx, 'tcx:'cx> {
 impl<'cx, 'tcx> OrphanChecker<'cx, 'tcx> {
     fn check_def_id(&self, item: &hir::Item, def_id: DefId) {
         if def_id.krate != LOCAL_CRATE {
-            span_err!(self.tcx.sess, item.span, E0116,
+            struct_span_err!(self.tcx.sess, item.span, E0116,
                       "cannot define inherent `impl` for a type outside of the \
                        crate where the type is defined; define and implement \
```
Comment (jonathandturner) [inline src/librustc_typeck/coherence/orphan.rs]: Can you remove the "; define and implement a trait or new type instead" from the error title?

**Comment by jonathandturner** [comment]
Thanks for the PR!

This looks really close.  I made a comment on the one part that still needs to be updated.  After that, [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) down to 1 commit, and I think we'll be ready to go.

**Comment by terrynsun** [comment]
Got it!

**Comment by jonathandturner** [comment]
Looks good!

@bors r+ rollup

**Pull request merged**